src/lib/history.ts

```diff
@@ -0,0 +1,29 @@
+import { WeeklyArticleLog, WeeklyArticleHistory } from '../components/ContributionsEpicTypes';
+
+export const getMondayFromDate = (date: Date): number => {
+    const day = date.getDay() || 7;
+    if (day !== 1) {
+        date.setHours(-24 * (day - 1));
+    }
+    return Math.floor(date.getTime() / 86400000);
+};
+
+export const getArticleViewCountForWeeks = (
+    history: WeeklyArticleHistory = [],
+    weeks: number,
+): number => {
+    const mondayThisWeek = getMondayFromDate(new Date());
+    const cutOff = mondayThisWeek - weeks * 7;
```

Comment:
Ahhh so the previous 4 weeks means the 4 weeks before the current week (starting on Monday), I hadn't quite understood that before.

Reply:
Exactly. It counts whole weeks up until the beginning of the current week.